app/Http/Transformers/CustomFieldsetsTransformer.php

```diff
@@ -0,0 +1,46 @@
+<?php
+namespace App\Http\Transformers;
+
+use App\Models\CustomFieldset;
+use App\Models\CustomField;
+use App\Models\AssetModel;
+use Illuminate\Database\Eloquent\Collection;
+use App\Helpers\Helper;
+use Gate;
+
+class CustomFieldsetsTransformer
+{
+
+    public function transformCustomFieldsets (Collection $fieldsets, $total)
+    {
+        $array = array();
+        foreach ($fieldsets as $fieldset) {
+            $array[] = self::transformCustomFieldset($fieldset);
+        }
+        return (new DatatablesTransformer)->transformDatatables($array, $total);
+    }
+
+    public function transformCustomFieldset (CustomFieldset $fieldset)
+    {
+        $fields = $fieldset->fields;
+        $models = $fieldset->models;
+        $modelsArray = array();
+        $totalModels = 0;
+        foreach ($models as $model)
+        {
+            $modelArray = array();
+            $modelArray["id"] = $model->id;
+            $modelArray["name"] = $model->name;
```

Comment:
It's definitely clearer to add the anonymous model to $modelsArray rather than create a $modelArray. I like that change, but I think you would still want to count during the loop since you have to know the length of the array for the transformer.  Maybe it wouldn't make a difference, but it seems a little more efficient to count during the loop rather than call a count function afterwards.